php/class-wp-cli.php

```diff
@@ -21,11 +48,77 @@ public function create_guest_authors( $args, $assoc_args ) {
 		global $coauthors_plus;
 
 		$defaults = array(
-				// There are no arguments at this time
-			);
+			'roles' => array( 'administrator', 'editor', 'author', 'contributor' ),
+			'batch-size' => 1000,
+		);
 		$this->args = wp_parse_args( $assoc_args, $defaults );
 
-		$users = get_users();
+		if ( is_array( $this->args['roles'] ) ) {
+			$role_whitelist = $this->args['roles'];
+		} else {
+			$role_whitelist = explode( ',', $this->args['roles'] );
+		}
+
+		if ( $this->args['batch-size'] > 0 && is_numeric( $this->args['batch-size'] ) ) {
+			$batch_size = (int) $this->args['batch-size'];
+		} else {
+			WP_CLI::error( __( 'Batch size must be an integer greater than zero.', 'co-authors-plus' ) );
+		}
+
+		foreach ( $role_whitelist as $role ) {
+			// Does this role exist in this instance?
+			if ( ! $GLOBALS['wp_roles']->is_role( $role ) ) {
+				WP_CLI::error( __( 'Role ', 'co-authors-plus' ) . $role . __( ' does not exist.', 'co-authors-plus' ) );
+			}
+
+			// Subscribers cannot be made guest authors en masse
+			// due to potential performance issues (sites can have millions of subscribers).
+			if ( 'subscriber' === $role ) {
+				WP_CLI::error( __( 'For performance reasons, subscribers cannot be made guest authors via wp-cli. Please use the admin interface.', 'co-authors-plus' ) );
+			}
+		}
+
+		$users_unfiltered_query = new WP_User_Query( array( 'number' => $batch_size ) );
```

Comment:
Do we get any benefits if we order by primary key?

Reply:
Not really, we're going to iterate over the entire array no matter what, so the order isn't important :)

Reply:
If we order by primary key, would it make the lookups and offsetting be faster?

Reply:
Turns out yes, yes it would: https://github.com/smistephen/Co-Authors-Plus/commit/7231625d3d42c3b3680c3ba036baa430944a4f64